contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,810 @@
+     */
+    function _utilizerBalanceStoredInternal(address account) internal view returns (uint256) {
+        /* Get utilizeBalance and utilizeIndex */
+        UtilizerStruct storage utilizeSnapshot = utilizerData[account];
+
+        /* If utilizeBalance = 0 then utilizeIndex is likely also 0.
+         * Rather than failing the calculation with a division by 0, we immediately return 0 in this case.
+         */
+        if (utilizeSnapshot.principal == 0) {
+            return 0;
+        }
+
+        /* Calculate new utilized balance using the utilize index:
+         *  recentUtilizeBalance = utilizer.principal * utilizeIndex / utilizer.utilizeIndex
+         */
+        uint256 principalTimesIndex = utilizeSnapshot.principal * utilizeIndex;
+        return principalTimesIndex / utilizeSnapshot.utilizeIndex;
+    }
+
+    /**
+     * @notice Calculates the exchange rate from the SD to the SDx token
+     * @dev This function does not accrue fee before calculating the exchange rate
+     * @return calculated exchange rate scaled by 1e18
+     */
+    function _exchangeRateStoredInternal() internal view virtual returns (uint256) {
+        if (cTokenTotalSupply == 0) {
+            /*
+             * If there are no tokens minted:
+             *  exchangeRate = initialExchangeRate
+             */
+            return DECIMAL;
+        } else {
+            /*
+             * Otherwise:
+             *  exchangeRate = (totalCash + totalUtilizedSD - totalFee) / totalSupply
+             */
+            uint256 poolBalancePlusUtilizedSDMinusReserves = getPoolAvailableSDBalance() +
+                totalUtilizedSD -
+                accumulatedProtocolFee;
+            uint256 exchangeRate = (poolBalancePlusUtilizedSDMinusReserves * DECIMAL) / cTokenTotalSupply;
+
+            return exchangeRate;
+        }
+    }
+
+    /// delete entry from delegatorWithdrawRequests mapping and in requestIdsByDelegatorAddress mapping
+    function _deleteRequestId(uint256 _requestId) internal {
+        delete (delegatorWithdrawRequests[_requestId]);
+        uint256 userRequestCount = requestIdsByDelegatorAddress[msg.sender].length;
+        uint256[] storage requestIds = requestIdsByDelegatorAddress[msg.sender];
+        for (uint256 i; i < userRequestCount; ) {
+            if (_requestId == requestIds[i]) {
+                requestIds[i] = requestIds[userRequestCount - 1];
+                requestIds.pop();
+                return;
+            }
+            unchecked {
+                ++i;
+            }
+        }
+        revert CannotFindRequestId();
+    }
+
+    function getUserData(address account) public view returns (UserData memory) {
+        address staderOracle = staderConfig.getStaderOracle();
+        uint256 sdPriceInEth = IStaderOracle(staderOracle).getSDPriceInETH();
+        uint256 accountUtilizePrev = _utilizerBalanceStoredInternal(account);
+        uint256 totalFeeSD = accountUtilizePrev - utilizerData[account].principal;
```

Comment:
this should be 
`accountUtilizePrev  - ISDCollateral(staderConfig.getSDCollateral()).operatorUtilizedSDBalance(account)`

Reply:
This is trying to get the total interest

Reply:
yeah, but the principal changes whenever the utilizer does utilise or repay operation, the new principal becomes the previous principal + interest accrued 
let's say a NO utilize 100 SD at 10%, principal is 100, if you compute interest after 1 year according to your logic it will give 10
but if NO again utilize 100 more SD after 1 yr, the principal will become 210 (110 + 100) and interest will be 0 according to your logic
but if use `ISDCollateral(staderConfig.getSDCollateral()).operatorUtilizedSDBalance(account)` it will only store the actual amount utilized at any point